### PR TITLE
feat(providers): ZITADEL provider

### PR DESCRIPTION
**Pull request by peintnermax**
## ☕️ Reasoning

Add [ZITADEL](https://zitadel.com) OIDC provider

## 🧢 Checklist

- [x] Documentation
- [x] Tests
- [x] Ready to be merged

## 📌 Resources

- [Security guidelines](../Security.md)
- [Contributing guidelines](../CONTRIBUTING.md)
- [Code of conduct](../CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

File: packages/next-auth/src/providers/zitadel.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    async profile(profile) {
+      return {
+        id: profile.sub,
+        name: profile.name,
+        firstName: profile.given_name,
+        lastName: profile.family_name,
+        email: profile.email,
+        image: profile.picture,
+        loginName: profile.preferred_username,
+      }
```
Comment (balazsorban44) [inline packages/next-auth/src/providers/zitadel.ts:53]:
```suggestion
      return {
        id: profile.sub,
        name: profile.name,
        email: profile.email,
        image: profile.picture,
      }
```

File: packages/next-auth/src/providers/zitadel.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    name: "ZITADEL",
+    type: "oauth",
+    version: "2",
+    wellKnown: `${issuer}/.well-known/openid-configuration`,
+    authorization: { params: { scope: "openid email profile" } },
+    idToken: true,
+    checks: ["pkce", "state"],
+    client: {
+      token_endpoint_auth_method: "none",
+    },
```
Comment (balazsorban44) [inline packages/next-auth/src/providers/zitadel.ts:43]:
```suggestion
```

From the docs: https://docs.zitadel.com/docs/apis/openidoauth/endpoints it does not sound like this is actually necessary. Can we just use the default,

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Thanks, added some suggestions

**Comment by peintnermax** [comment]
Thanks! Applied and tested your suggestions 👍

**Review by balazsorban44 — APPROVED** [review]

**Pull request merged**